src/m365/teams/commands/chat/chat-message-send.ts

```diff
@@ -0,0 +1,244 @@
+    const currentUserEmail = Utils.getUserNameFromAccessToken(Auth.service.accessTokens[this.resource].accessToken).toLowerCase();    
+    const existingChats = await this.findExistingGroupChatsByMembers([currentUserEmail, ...userEmails]);
+
+    if (existingChats && existingChats.length > 0) {
+      if (existingChats.length > 1) {
+        const disambiguationText = existingChats.map(c => {
+          return `- ${c.id}${c.topic && " - "}${c.topic} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()}`;
+        }).join(os.EOL);
+
+        throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+      } 
+      else {
+        return existingChats[0].id;
+      }
+    }
+
+    const chat = await this.createConversation([currentUserEmail, ...userEmails]);
+    return chat.id;
+  }
+
+  private async getChatIdByName(chatName: string): Promise<string> {    
+    const existingChats = await this.findExistingGroupChatsByTopic(chatName);
+        
+    if (!existingChats || existingChats.length === 0) {          
+      throw new Error("No chat conversation was found with this name.");
+    }
+    else if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+    else {
+      const disambiguationText = existingChats.map(c => { 
+        const memberstring = (c.members as ConversationMember[]).map(m => (m as AadUserConversationMember).email).join(", ");
+        return `- ${c.id} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()} - ${memberstring}`;
+      }).join(os.EOL);
+
+      throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+    }
+  }
+
+  // This Microsoft Graph API request throws an intermittent 404 exception, saying that it cannot find the principal.
+  // The same behavior occurs when creating the conversation through the Graph Explorer.
+  // It seems to happen when the userEmail casing does not match the casing of the actual UPN. 
+  // When the first request throws an error, the second request does succeed. 
+  // Therefore a retry-mechanism is implemented here. 
+  private async createConversation(memberEmails: string[], retried: number = 0): Promise<Chat> {
+    try {
+      const jsonBody = {
+        chatType: memberEmails.length > 2 ? "group" : "oneOnOne",
+        members: memberEmails.map(email => {
+          return {
+            "@odata.type": "#microsoft.graph.aadUserConversationMember",
+            roles: ["owner"],
+            "user@odata.bind": `https://graph.microsoft.com/v1.0/users/${email}`
+          };
+        })
+      };
+      
+      const requestOptions = this.getRequestOptions(`${this.resource}/v1.0/chats`, jsonBody);
+      return await request.post<Chat>(requestOptions);    
+    } 
+    catch(err: any) {
```

Comment:
Since we're working with the `err` object in the code, let's use its type rather than `any` to use design-time type safety

Reply:
Yeah, I did try that, but was hindered by the linter:

`Catch clause variable type annotation must be 'any' or 'unknown' if specified.ts(1196)`

Doesn't make sense to me, but I left it at 'any' after that.